frontend/src/app/(vendor)/partner/settings/page.tsx

```diff
@@ -0,0 +1,30 @@
+import { redirect } from 'next/navigation';
+import { getCurrentUserAction } from '@/lib/auth/actions/getUser';
+import { getVendorForCurrentUser } from '@/features/profile/manage/api/getVendorForCurrentUser';
+import { Settings } from '@/features/settings/components/Settings';
+
+export default async function VendorSettingsPage() {
+  // Check authentication
+  const currentUser = await getCurrentUserAction();
+
+  if (!currentUser || !currentUser.userId) {
+    redirect('/partner/login?redirect=/partner/manage');
```

Comment:
should this redirect to the settings page since that's the original page they're trying to view